### DISCUSSION TITLE
typeof 연산자는 무엇이고, 어떻게 사용하나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#30] [데이터 타입]
- typeof 연산자는 무엇이고, 어떻게 사용하나요?
- typeof 연산자와 instanceof 연산자의 차이점은 무엇인가요?

**Comment by pipisebastian** [comment]
>typeof 연산자는 무엇이고, 어떻게 사용하나요?

변수의 자료형을 반환합니다. string, number등을 반환하며, 찾고자하는 변수의 앞에서 사용합니다. `typeof num`

>typeof 연산자와 instanceof 연산자의 차이점은 무엇인가요?

typeof은 해당하는 타입을 문자열로 반환하는 반면, instanceof는 특정 클래스에 속하는 객체인지 여부(true/false)를 반환합니다.

**Comment by publdaze** [comment]
> - typeof 연산자는 무엇이고, 어떻게 사용하나요?
- typeof 연산자는 변수나 값의 기본 데이터 타입 반환
> - typeof 연산자와 instanceof 연산자의 차이점은 무엇인가요?
- typeof는 주로 원시 데이터 타입을 확인할 때 사용
- instanceof 연산자는 객체가 특정 클래스 또는 생성자 함수의 인스턴스인지 여부를 확인, 객체가 특정 클래스나 생성자의 인스턴스인지 확인할 때 사용됩니다.
